lib/common/socket/evloop.c.h

```diff
@@ -535,6 +535,38 @@ static void run_pending(h2o_evloop_t *loop)
     }
 }
 
+void h2o_evloop_destroy(h2o_evloop_t *loop)
+{
+    struct st_h2o_evloop_socket_t *sock;
+    //dispose all socket
+    while (loop->_pending_as_server != NULL || loop->_pending_as_client != NULL) {
+        while ((sock = loop->_pending_as_client) != NULL) {
+            loop->_pending_as_client = sock->_next_pending;
+            sock->_next_pending = sock;
+            do_dispose_socket((h2o_socket_t *)sock);
+        }
+        if ((sock = loop->_pending_as_server) != NULL) {
+            loop->_pending_as_server = sock->_next_pending;
+            sock->_next_pending = sock;        
+            do_dispose_socket((h2o_socket_t *)sock);
+        }
+    }
+      //looks like no need to free  _timeouts 
+      //h2o_timeout__do_dispose(loop,  loop->_timeouts);
+    
+    //now all socket are disposedand and placed in linked list statechanged
+    //we can freeing memory in cycle by next_statechanged,
+    while((struct st_h2o_evloop_socket_t*)(sock=loop->_statechanged.head->_next_statechanged) != NULL){
+
+        free(loop->_statechanged.head);
+        loop->_statechanged.head=loop->_statechanged.head->_next_statechanged;
```

Comment:
Yes, i make this stupid mistake, when i really sleeping.
Thank you for pointing me to this. 
I think with last [commit](https://github.com/h2o/h2o/pull/1200/commits/6dcc57ecaddd4196a967cee3f4bbf0868a56eaab)  i fixed it.
But something wrong with travis-ci. It does not compile my code,because can not find some ubuntu packages: Err http://ppa.launchpad.net precise/main amd64 Packages 404  Not Found.
On my computer everything compiling fine